src/system/DurationTimer.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <string>
+
+#include <system/SystemConfig.h>
+
+/**
+ * @file
+ * 
+ *  Uses SystemClock.h HAVE_CLOCK_GETTIME to determine board type i.e. esp32 or nRF etc.
+ *  CHIP_DEVICE_USES_SYS_TIME flag indicates nRF board whose clock implementation is available in Syste,Clock.h.
+ *  CHIP_DEVICE_USES_TIME_H flag for esp32 uses a time.h timespec
+ */
+
+#if (CHIP_SYSTEM_CONFIG_PLATFORM_PROVIDES_TIME && (CHIP_SYSTEM_CONFIG_USE_POSIX_TIME_FUNCTS || CHIP_SYSTEM_CONFIG_USE_SOCKETS))
+#define CHIP_DEVICE_USES_SYS_TIME 1
+#else // ! CHIP_SYSTEM_CONFIG_PLATFORM_PROVIDES_TIME etal
+#define CHIP_DEVICE_USES_TIME_H 1
+#endif // CHIP_SYSTEM_CONFIG_PLATFORM_PROVIDES_TIME etal
+
+
+#if CHIP_DEVICE_USES_SYS_TIME
+/**
+ * Use SystemClock/clock-gettime  
+ * */
+#include <system/SystemClock.h>
+#endif
+
+#if CHIP_DEVICE_USES_TIME_H
+/**
+ * SystemClock/clock-gettime not supported therefore use time.h
+ * */
+#include <time.h>
+#endif
+
+using namespace std;
+
+namespace chip {
+namespace timing {
+
+class DurationTimer
+{
+#ifdef CHIP_DEVICE_USES_SYS_TIME
+/**
+ * SystemClock/clock-gettime supported therefore timeval struct.
+ * */
+private:
+    string toTimeStr(timeval * time);
+
+protected:
+    timeval t1;
+    timeval t2;
+    string label;
+#endif
+
+#ifdef CHIP_DEVICE_USES_TIME_H
+/**
+ * clock-SystemClock/clock-gettime not supported therefore use timespec struct
+ * */
+private:
+    string toTimeStr(timespec * time);
+
+protected:
+    timespec t1;
+    timespec t2;
```

Comment:
Why do we need these members?  We should just store the actual monotonic timestamps, I would think, instead of storing things that look like wall-clock times.